src/Distribution/Nixpkgs/Meta.hs

```diff
@@ -85,6 +92,54 @@ nixpkgsPlatformFromCabal (Platform arch os) = "\"" ++ nixArch ++ "-" ++ nixOs ++
             OSX -> "darwin" -- rendered as osx by default
             _   -> CabalPretty.prettyShow os
 
+-- | Obtain a 'NixpkgsPlatform' from a string representation intended for config
+--   files.
+--
+--   * Every string starting with @lib.platforms.@ or @platforms.@ is
+--     parsed into 'NixpkgsPlatformGroup'.
+--   * All other strings are attempted to be interpreted as a system tuple of
+--     the form @ARCH-KERNEL@ and parsed into 'NixpkgsPlatformSingle'.
+--
+--   If none of these formats match the input 'String', 'Nothing' is returned.
+--   A 'Just' result thus only indicates that the format of the platform is
+--   sound — 'nixpkgsPlatformFromString' does /not/ check if the parsed platform
+--   actually exists.
+--
+--   __Note__: 'nixpkgsPlatformFromString' is /not/ the inverse operation for
+--   'NixpkgsPlatform'\'s 'Pretty' instance. It is not intended for parsing Nix
+--   expressions.
+--
+--   >>> nixpkgsPlatformFromString "x86_64-netbsd"
+--   Just (NixpkgsPlatformSingle (Platform X86_64 NetBSD))
+--   >>> nixpkgsPlatformFromString "platforms.riscv"
+--   Just (NixpkgsPlatformGroup (Identifier "riscv"))
+--   >>> nixpkgsPlatformFromString "garbage"
+--   Nothing
+nixpkgsPlatformFromString :: String -> Maybe NixpkgsPlatform
+nixpkgsPlatformFromString s = platformGroup <|> singlePlatform
+  where platformGroup = do
+          -- also accept "platform." as prefix to save some typing
+          name <- stripPrefix "lib.platforms." s <|> stripPrefix "platforms." s
+          Just $ NixpkgsPlatformGroup (ident # name)
+
+        -- we accept anything as a platform tuple which is two non-empty
+        -- strings separated by a '-'
+        splitPlatformTuple = filter (not . null) . splitOn "-"
+        singlePlatform =
```

Comment:
I'm currently looking to split out the `singlePlatform` parser into its own function and expose it. This would allow reuse in `cabal2nix`/`hackage2nix` which need to parse such platforms on the command line.

Reply:
Done.